src/main.rs

```diff
@@ -260,7 +294,13 @@ fn main() {
         // The main thread just walks and filters the directory tree.  Symlinks
         // are uninteresting and ignored.
         for root in roots {
-            for dir_entry in WalkDir::new(root).follow_links(false) {
+            let walkdir = 
+                if non_recursive {
+                    WalkDir::new(root).max_depth(1).follow_links(false)
+                } else {
+                    WalkDir::new(root).min_depth(0).follow_links(false)
```

Comment:
Is the `min_depth` needed? It wasn't there before...

Reply:
Hm, perhaps know. Have to test.